File: Stack/921.minimum-add-to-make-parentheses-valid.go
package leetcode

// @leet start
func minAddToMakeValid(s string) int {
	stack := make([]rune, 0, len(s))
	for _, r := range s {
		switch r {
		case '(':
			stack = append(stack, r)
		case ')':
			if len(stack) > 0 && stack[len(stack)-1] == '(' {
				stack = stack[:len(stack)-1]
				break
			}
			stack = append(stack, r)
		}
	}
	return len(stack)
}

// @leet end

File: Stack/921.minimum-add-to-make-parentheses-valid.py
# @leet start
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        st = []
        for c in s:
            if c == "(":
                st.append(c)
            else:
                if st and st[-1] == "(":
                    st.pop()
                else:
                    st.append(c)
        return len(st)


# @leet end

